File: Daily Challanges/2026-06-June/2026-06-17.js
function cast(spells) {
  const score = {
    f: [3, "D"],
    l: [3, "D"],
    i: [2, "C"],
    w: [2, "C"],
    h: [1, "R"],
    s: [1, "R"],
  };

  let total = 0;
  let multiplier = 1;
  let prevCategory = null;

  for (const spell of spells) {
    const [base, category] = score[spell];

    if (prevCategory !== null) {
      if (category !== prevCategory) {
        multiplier++;
      } else {
        multiplier = 1;
      }
    }

    total += base * multiplier;
    prevCategory = category;
  }

  return total;
}

console.log(cast("fihwl"));
console.log(cast("lwswfi"));
console.log(cast("wislhfl"));
console.log(cast("sihwlih"));
console.log(cast("wishlfihwslwifihl"));

File: Daily Challanges/2026-06-June/2026-06-17.py
def cast(spells):
    score = {
        "f": (3, "D"),
        "l": (3, "D"),
        "i": (2, "C"),
        "w": (2, "C"),
        "h": (1, "R"),
        "s": (1, "R"),
    }

    total = 0
    multiplier = 1
    prev_category = None

    for spell in spells:
        base, category = score[spell]

        if prev_category is not None:
            if category != prev_category:
                multiplier += 1
            else:
                multiplier = 1

        total += base * multiplier
        prev_category = category

    return total


print(cast("fihwl"))
print(cast("lwswfi"))
print(cast("wislhfl"))
print(cast("sihwlih"))
print(cast("wishlfihwslwifihl"))